corehq/apps/geospatial/static/geospatial/js/geospatial_map.js

```diff
@@ -72,17 +74,34 @@ hqDefine("geospatial/js/geospatial_map", [
                 }
             });
 
+            function changeCaseMarkerColor(selectedCase, newColor) {
+                const caseId = selectedCase.case_id;
+                const caseMarker = Object.entries(caseMarkers).find(([key]) => key === caseId);
+                if (caseMarker) {
+                    const [caseId, marker] = caseMarker;
+                    let element = marker.getElement();
+                    let svg = element.getElementsByTagName("svg")[0];
+                    let path = svg.getElementsByTagName("path")[0];
+                    path.setAttribute("fill", newColor);
+                } else {
+                    console.error("No marker was found to be associated with a case!");
```

Comment:
If this error occurred, is this something that would block the user from interacting with the map? If so, then I'm wondering if it might be useful to give a message to the user in the form of an alert banner as well (just so that they don't wonder what's happening when their interactions aren't working).

Reply:
This would only have happend because we had two lists containing cases: 1. the list of cases and 2. a mapping of a case to its marker. I realized this is a stupid way to go, so I refactored the code to add the marker obj on the case obj itself. Now it's also way easier and more intuitive to work with a case and its marker